v2/pkg/subscraping/sources/shodan/shodan.go

```diff
@@ -75,10 +76,13 @@ func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Se
 			}
 
 			for _, data := range response.Subdomains {
-				results <- subscraping.Result{
-					Source: s.Name(), Type: subscraping.Subdomain, Value: fmt.Sprintf("%s.%s", data, domain),
+				value := fmt.Sprintf("%s.%s", data, response.Domain)
+				if strings.HasSuffix(value, domain) {
```

Comment:
```suggestion
				suffix := fmt.Sprintf(".%s", strings.TrimPrefix(domain, "."))
				if strings.HasSuffix(value, suffix) {
```

Reply:
This should address your concern.

Reply:
I'll debug and validate this, thanks !
I think it might filter out the original api.example.com though, if we filter for .api.example.com, the subdomain cannot be underneath itself.

Reply:
This is getting more complicated than I initially thought, and it seems like this is just a temporary workaround. Moving forward, we could use the [`x/net/publicsuffix`](https://pkg.go.dev/golang.org/x/net/publicsuffix) pkg to properly compare suffixes by stripping out the root domain (eTLD+1) correctly.

Reply:
<img width="1346" alt="image" src="https://github.com/user-attachments/assets/62e1a262-9cab-4bfd-9c93-66ded30d6de4">
Ok I think another thing that is confusing is that we do have a filtering in place that removes any subdomain that does not fit under the supplied subdomain. Weirdly though, `analysiscenter.veracode.com` appears on my non-fixed version of subfinder...

Reply:
Nevermind, I think `crtsh` is returning `"*.analysiscenter.veracode.com"` and therefore this passes the validation, so you find the supplied domain as a subdomain of itself

Reply:
I think crtsh is an edge case, given the suffix check is done on all sources at the end, I think shodan should NOT include the subdomain itself if it's found in the list of subdomains, so we're ok here :)